susemanager-utils/susemanager-sls/src/modules/mgr_caasp_manager.py

```diff
@@ -0,0 +1,347 @@
+        'stderr': salt.utils.stringutils.to_str(skuba_proc.stderr),
+        'success': not skuba_proc.process.returncode,
+        'retcode': skuba_proc.process.returncode,
+    }
+    return ret
+
+
+def cluster_init(name,
+                 cluster_path,
+                 target,
+                 cloud_provider=None,
+                 strict_capability_defaults=False,
+                 verbosity=None,
+                 timeout=DEFAULT_TIMEOUT,
+                 **kwargs):
+
+    cmd_args = "cluster init --control-plane {} {}".format(target, name)
+
+    if cloud_provider:
+        cmd_args += " --cloud-provider {}".format(cloud_provider)
+    if strict_capability_defaults:
+        cmd_args += " --strict-capability-defaults"
+    if verbosity:
+        cmd_args += " --verbosity {}".format(verbosity)
+
+    skuba_proc = _call_skuba(cluster_path, cmd_args)
+    if skuba_proc.process.returncode != 0:
+        error_msg = "Unexpected error {} at skuba when initializing the cluster: {}".format(
+                skuba_proc.process.returncode,
+                salt.utils.stringutils.to_str(skuba_proc.stderr))
+        log.error(error_msg)
+
+    ret = {
+        'stdout': salt.utils.stringutils.to_str(skuba_proc.stdout),
+        'stderr': salt.utils.stringutils.to_str(skuba_proc.stderr),
+        'success': not skuba_proc.process.returncode,
+        'retcode': skuba_proc.process.returncode,
+    }
+    return ret
+
+
+def master_bootstrap(node_name,
+                     skuba_cluster_path,
+                     target,
+                     ignore_preflight_errors=None,
+                     port=None,
+                     sudo=None,
+                     user=None,
+                     verbosity=None,
+                     timeout=DEFAULT_TIMEOUT,
+                     **kwargs):
+
+    cmd_args = "node bootstrap --target {} {}".format(target, node_name)
+
+    if ignore_preflight_errors:
+        cmd_args += " --ignore-preflight-errors {}".format(ignore_preflight_errors)
+    if port:
+        cmd_args += " --port {}".format(port)
+    if sudo:
+        cmd_args += " --sudo"
+    if user:
+        cmd_args += " --user {}".format(user)
+    if verbosity:
+        cmd_args += " --verbosity {}".format(verbosity)
+
+    skuba_proc = _call_skuba(skuba_cluster_path, cmd_args)
+    if skuba_proc.process.returncode != 0:
+        error_msg = "Unexpected error {} at skuba when bootstrapping the node: {}".format(
+                skuba_proc.process.returncode,
+                salt.utils.stringutils.to_str(skuba_proc.stderr))
+        log.error(error_msg)
+
+    ret = {
+        'stdout': salt.utils.stringutils.to_str(skuba_proc.stdout),
+        'stderr': salt.utils.stringutils.to_str(skuba_proc.stderr),
+        'success': not skuba_proc.process.returncode,
+        'retcode': skuba_proc.process.returncode,
+    }
+    return ret
+
+
+def create_cluster(cluster_name,
+                   cluster_path,
```

Comment:
@mateiw actually those two parameters are not representing the same:

- `cluster_path` would be something like `/srv/my-clusters/`. The path where the cluster directory will be created.
- `skuba_cluster_path` is where your particular cluster is stored: i.a. `/srv/my-clusters/my-first-cluster/`

Since they sound similar, maybe I should change `cluster_path` to something else more descriptive to avoid the confusion. I'll go maybe with `base_path`